docs/docs/general/02-Getting Started/write-your-first-automation.md

```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 3
+---
+
+# Write Your First Automation
+
+:::tip
+What you will learn
+
+- Initialize your first automation suite
+- Configure your credentials (workspace id and access token)
+- Understand the generated automation case
+- Run your first automation
+- Troubleshooting any issues
+- Where to go next
+:::
+
+## Quickstart
+
+## Initialization
+
+To create your first automation suite, enter
+
+```shell
+npx askui init
+```
+
+<details>
+  <summary>This will create</summary>
+
+-   a `tsconfig.json`: [a json file specifying the root files and the compiler options required to compile the project](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html),
+-   a folder called `test` which includes:
+    -   `test/my-first-askui-test-suite.test.ts`: an example test with askui,
+    -   a folder called `helper` which contains the `jest.setup.ts` file for setting up the test environment
+    -   a `jest.config.ts` configuration file
+
+</details>
+
+### Configuration
+
+Generate credentials through the [user portal](https://app.v2.askui.com/) (usage is free!). Please see the [documentation](https://docs.askui.com/docs/general/askui%20User%20Portal/signup) on signing up and generating credentials.
+
+Then, go to your `helper/jest.setup.ts` and add the configuration for your `workspace id` and your `access token` to the _UiControlClient_.
+
+```typescript
+ aui = await UiControlClient.build({
+    credentials: {
+      workspaceId: '<your workspace id>',
+      token: '<your access token>',
+    }
+  });
+```
+
+### Understand the Generated Code
+
+### Run
+
+:::info
+Before executing the automation, open `test/my-first-askui-test-suite.test.ts` on your main display. This file contains the text on **line 12** for which the AI will search.
+
+Some users have reported instability running automation on macOS with external displays and/or virtual desktops. If you experience similar issues, please disconnect external displays and close virtual desktops, or see [documentation on running automation in Docker](../04-Continuous%20Integration/askui-ui-controller-docker-images.md).
+:::
+
+To execute the automation suite, enter
+
+```shell
+npx jest test/my-first-askui-test-suite.test.ts --config ./test/jest.config.ts 
```

Comment:
Can we add a script for that inside the `package.json` when running `npx askui init`?

Reply:
I already thought about that as it would make the life of the user easier. But only if you use Jest. So I was hesitant to add it as a default.

Reply:
Moved to future TODO; I don't think it is a problem to add a script to the `package.json`. For now we only support jest when initializing so this is fine. Later, when we add support for more frameworks, we can make the script dependent upon which framework we are using. So no problem from my perspective here. Only thing to consider is that the script name should be unique, maybe something like `test:askui`